apps/ledger-live-desktop/src/newArch/hooks/useStake.ts

```diff
@@ -0,0 +1,151 @@
+import { useCallback, useContext, useMemo } from "react";
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+import { liveAppContext as remoteLiveAppContext } from "@ledgerhq/live-common/platform/providers/RemoteLiveAppProvider/index";
+import { LiveAppRegistry } from "@ledgerhq/live-common/platform/providers/RemoteLiveAppProvider/types";
+import { liveAppContext as localLiveAppProviderContext } from "@ledgerhq/live-common/wallet-api/LocalLiveAppProvider/index";
+import { LiveAppManifest, Loadable } from "@ledgerhq/live-common/platform/types";
+import { appendQueryParamsToManifestURL } from "@ledgerhq/live-common/platform/utils/appendQueryParamsToManifestURL";
+import type { Account, AccountLike, TokenAccount } from "@ledgerhq/types-live";
+import {
+  getAccountCurrency,
+  getAccountSpendableBalance,
+  isTokenAccount,
+} from "@ledgerhq/coin-framework/account/index";
+import { accountToWalletAPIAccount } from "@ledgerhq/live-common/wallet-api/converters";
+import { WalletState } from "@ledgerhq/live-wallet/store";
+import { deriveAccountIdForManifest } from "@ledgerhq/live-common/platform/utils/deriveAccountIdForManifest";
+
+const getRemoteLiveAppManifestById = (
+  appId: string,
+  liveAppRegistry: Loadable<LiveAppRegistry>,
+) => {
+  if (!liveAppRegistry.value || !appId) {
+    return undefined;
+  }
+
+  return (
+    liveAppRegistry.value.liveAppFilteredById[appId] || liveAppRegistry.value.liveAppById[appId]
+  );
+};
+
+export function useStake() {
+  const { params } = useFeature("stakePrograms") || {};
+  const enabledCurrencies = useMemo(() => params?.list ?? [], [params?.list]);
+  const redirectsMap = useMemo(
+    () => new Map(Object.entries(params?.redirects ?? [])),
+    [params?.redirects],
+  );
+  const partnerSupportedAssets = useMemo(() => Array.from(redirectsMap.keys()), [redirectsMap]);
+  const getLocalLiveAppManifestById = useContext(
+    localLiveAppProviderContext,
+  ).getLocalLiveAppManifestById;
+  const remoteLiveAppRegistry: Loadable<LiveAppRegistry> = useContext(remoteLiveAppContext).state;
+
+  const getManifest = useCallback(
+    (platformId: string) => {
+      const localManifest = getLocalLiveAppManifestById(platformId);
+      const remoteManifest = getRemoteLiveAppManifestById(platformId, remoteLiveAppRegistry);
+      const manifest: LiveAppManifest | undefined = localManifest || remoteManifest;
+      return manifest;
```

Comment:
We could return the manifest directly if you want but not a big deal to keep it like that
Careful committing the suggestion directly as it will leave the `LiveAppManifest` type unused in this file
```suggestion
      return localManifest || remoteManifest;
```